src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object}  currentState
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+      states.push({ ...currentState });
+    }
+
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+      states.push({ ...currentState });
+    }
+
+    if (action.type === 'removeProperties') {
```

Comment:
The current implementation lacks a default case for error handling in the action type checks. Refer to the checklist item: 'switch/case should always have default case for error handling.'